### ISSUE TITLE
Scenario Outline + restartBrowserBetweenTests:true opens multiple empty browsers

### ISSUE THREAD

**Issue by manu7691** [protractor-cucumber-framework/protractor-cucumber-framework#100]
I found an issue when I'm trying to run a feature with scenario outline with several examples.

I'm using Protractor 5.1.2, Cucumber 1.3.3, Protractor-Cucumber-Framework 4.0.4.

On the config file, I have restartBrowserBetweenTests:true.

```gherkin
Given I have a feature with scenario outline and several examples
When I run tests
Then I see one browser running the first example
And I see a new empty browser each time a step from example is executed.
```

Any ideas? 

Thanks in advance

**Comment by darrinholst** [comment]
I think there's something going on with `restartBrowserBetweenTests` that's causing this as well as #88. I pushed a reduce test case [here](https://github.com/dholst/protractor-cucumber-framework-example/tree/scenario-outlines-with-browser-restart) if anyone wants to try and diagnose it. The behavior I'm seeing is the first example always fails with...

> Error: Error while waiting for Protractor to sync with the page: "window.angular is undefined.  This could be either because this is a non-angular page or because your test involves client-side navigation, which can interfere with Protractor's bootstrapping.  See http://git.io/v4gXM for details"

...and leaves that window open. The rest of the examples pass.

**Comment by pawelus** [comment]
@darrinholst we noticed problems when we switched from Protractor 4 to Protractor 5 so it's a change in Protractor that causes erroneous behaviour in cucumber-protractor-framework.

How various protractor-cucumber-framework versions behave in this scenario:
- 0.6.0 - no extra windows are spawned but only first 2 examples work fine. 3rd example will reuse the window from the 2nd one without cleaning the session so tests fail
- 1.0.2 - windows work as they should. new window open for every example, no extra windows spawned but with `shardTestFiles: true` on Protractor process hangs forever after test causing our Jenkins jobs to hang
- 2.3.0 - windows work as they should. new window open for every example, no extra windows spawned - haven't used this one on Jenkins but might try if you want
- 3.1.2 - spawns extra window after every step
- 4.0.4 - spawns extra window after every step

So some change between 2.3.0 and 3.1.2 is causing this window spawning problem
